modules/solid_mechanics/src/bcs/DirectDirichletBCBase.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DirectDirichletBCBase.h"
+#include "NonlinearSystemBase.h"
+
+InputParameters
+DirectDirichletBCBase::validParams()
+{
+  InputParameters params = NodalBC::validParams();
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  return params;
+}
+
+DirectDirichletBCBase::DirectDirichletBCBase(const InputParameters & parameters)
+  : NodalBC(parameters),
+    _preset(false),
+    _nonlinear_implicit_system(dynamic_cast<NonlinearImplicitSystem *>(&_sys.system())),
+    _mass_matrix()
+{
+}
+
+void
+DirectDirichletBCBase::initialSetup()
+{
+  _mass_matrix = &_nonlinear_implicit_system->get_system_matrix();
```

Comment:
this confuses me. In your test you have an extra tagged matrix called "mass" which is what the `MassMatrix` kernel adds into. What is in the system matrix? It seems like it's not the mass. What I would expect to see is the `_mass_matrix` corresponding to the tagged mass matrix, not the system matrix. Else the `_mass_matrix` data member should have some different name

Reply:
If you still need to access the system matrix in this class, I would do it with `_sys.system().get_matrix("System Matrix")`. In that case you will get a map find error if there is no system matrix which could happen if the system is not an `ImplicitSystem`. Using `_sys.system().get_matrix("System Matrix")` also has the conceptual benefit of working with both `NonlinearImplicitSystem` and `LinearImplicitSystem`.

Reply:
The logic here is that `CentralDifferenceDirect::performExplicitSolve` compute the mass matrix and stores it in `_nonlinear_implicit_system->get_system_matrix()` before the residual is calculated. `DirectDirichletBCBase.C` is retrieving that matrix. 
I don't believe the system matrix is ever written by anything else. This is the same logic currently used in `ActuallyExplicitEuler`.

Reply:
I believe this is resolved here, by tagging the system matrix instead of generating a new tag.
[9ece25d](https://github.com/idaholab/moose/pull/28175/commits/9ece25dbf1c22cb308158e04798e1903c05d45dc)

Reply:
agreed!